.github/workflows/bin-ci.yml

```diff
@@ -4,22 +4,38 @@ on:
     branches: '**'
 
 env:
-  GO_VERSION: '1.17'
   CONSUL_LICENSE: ${{ secrets.CONSUL_LICENSE }}
 
 jobs:
+  get-go-version:
+    name: "Determine Go toolchain version"
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.get-go-version.outputs.go-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine Go version
+        id: get-go-version
+        # We use .go-version as our source of truth for current Go
+        # version, because "goenv" can react to it automatically.
+        run: |
+          echo "Building with Go $(cat .go-version)"
+          echo "::set-output name=go-version::$(cat .go-version)"
```

Comment:
I copied this over from the build.yml to read go version from `.go-version` instead of using an environment variable.

Reply:
Ya, this is way better than having it specified in two places.